### ISSUE TITLE
macos shows error when running binaries (missing notarization)

### ISSUE THREAD

**Issue by apricote** [hetznercloud/cli#828]
### TL;DR

Our release binaries show an error when executed on macos.

### Expected behavior

The binaries should run without issues on macos.

### Observed behavior

macos shows an error screen because the binary was not notarized.

![macos-cli-notarization](https://github.com/user-attachments/assets/2d978971-065d-4ce1-8396-7845bf3f2da2)

### Minimal working example

0. Be a macos user
1. Download release binary
2. Run binary
3. Observe error

If you can not reproduce this, @LKaemmerling can reproduce locally.

### Additional information

Probably changed with https://github.com/hetznercloud/cli/commit/703f535444ac6e6dfa46470262c7bfd375f868d4

**Comment by jooola** [comment]
Possible alternative to rcodesign if needed: https://github.com/anchore/quill
